backend/app/services/implementations/residents_service.py

```diff
@@ -120,6 +120,50 @@ def get_residents(
                     .with_entities(Residents, Buildings.name.label("building"))
                     .all()
                 )
+            elif filters:
+                resident_id = filters.get("resident_id")
+                building_id = filters.get("building_id")
+                date_left = None
+                date_joined = None
+
+                if filters.get("date_range") is not None:
+                    date_joined, date_left = filters.get("date_range")
+                    if date_joined is not None:
+                        date_joined = datetime.strptime(
+                            date_joined, "%Y-%m-%d"
+                        ).replace(hour=0, minute=0)
+                    if date_left is not None:
+                        date_left = datetime.strptime(date_left, "%Y-%m-%d").replace(
+                            hour=0, minute=0
+                        )
+
+                residents_results = Residents.query.join(
```

Comment:
This probably can be moved out of the if-else statements since it is always used.